app/celery/research_mode_tasks.py

```diff
@@ -123,7 +124,18 @@ def firetext_callback(notification_id, to):
 def create_fake_letter_response_file(self, reference):
     now = datetime.utcnow()
     dvla_response_data = '{}|Sent|0|Sorted'.format(reference)
-    upload_file_name = 'NOTIFY-{}-RSP.TXT'.format(now.strftime('%Y%m%d%H%M%S'))
+
+    # try and find a filename that hasn't been taken yet - going back in time 60 seconds
+    for i in range(30):
```

Comment:
I think from what we've seen functional tests can hit this task a few times in a second. So if more than one task hits this loop they'll be trying the same numbers, which means we'll be repeating the same check and there could be a race condition between the check and the file creation.

Maybe we could shuffle the array to try slots in the last 30 seconds in random order.